draft-yasskin-http-origin-signed-responses.md

```diff
@@ -1342,24 +1304,36 @@ signed exchange.
 
 Specification: This document
 
-## Internet Media Type application/http-exchange+cbor
+## Internet Media Type application/signed-exchange
 
 Type name:  application
 
-Subtype name:  http-exchange+cbor
+Subtype name:  signed-exchange
 
-Required parameters:  N/A
+Required parameters:
+
+* v: An integer denoting the version of the file format. The version defined in
+  this specification is `1`. When used with the `Accept` header field (Section
+  5.3.1 of {{!RFC7231}}), this parameter can be a hyphen (-)-separated range of
+  version numbers, or a comma (,)-separated list of such ranges or individual
+  version numbers. The server is then expected to reply with a resource using a
+  particular version within those ranges.
```

Comment:
That seems a lot of complexity to put on the receiver. Why not just define application/signed-exchange-v1, application/signed-exchange-v2, etc. You don't need to parse out all this complex versioning information. We use a lot more bytes but header compression is nice, and is this format really going to have to change that much?

You might be able to make this extensible by saying the headers block is allowed to have unknown keys that you just ignore. This is how TLS does a lot of things. Though then you need to be sure that extensions are all safe for unknowing receivers to ignore.

Reply:
Hopefully the format won't change much once it's an RFC, but it is going to change until then. I think I talked to @sleevi about whether to make the headers block (especially the non-signed block) extensible, and he liked making it really simple, hence 21692e13a3b505292fbb248883ba368c4a2f178c.

I think I can remove ranges from the version numbers, since most likely a client will only support 2 versions at a time.